### DISCUSSION TITLE
[FR] Optional root `<template>` when the templating language is html

### DISCUSSION THREAD

**Discussion by wenfangdu** [vuejs/rfcs#525] [General]
### What problem does this feature solve?

Currently, when authoring components in SFC, we have to add a root `<template>` tag:

```vue
<script lang="ts" setup>
  // ...
</script>

<template>
  <div>Content</div>
</template>
```

Would be nice to be able to omit the root `<template>` tag when the templating language is html:

```vue
<script lang="ts" setup>
  // ...
</script>

<div>Content</div>
```

The above syntax is closer to writing html files, therefore more natural and intuitive, [Svelte does this by default](https://svelte.dev/docs#template-syntax), would be great if Vue supports it too.

### What does the proposed API look like?

```vue
<script lang="ts" setup>
  // ...
</script>

<div>Content</div>
```

**Comment by edison1105** [comment]
this is planned in Vapor mode

**Comment by wenfangdu** [comment]
@edison1105 Where is it documented and how can I try it?

**Comment by sxzz** [comment]
Not available for now. It's just a thought.

**Comment by Shyam-Chen** [comment]
- https://github.com/vitejs/vite/issues/17#issuecomment-638847357
- [@volar/vue-language-plugin-pug](https://github.com/vuejs/language-tools/tree/master/packages/vue-language-plugin-pug)

**Comment by wenfangdu** [comment]
@Shyam-Chen This issue has nothing to do with `pug`, and the optional root `<template>` should be enabled only when the templating language is html.
